practice-two/src/components/FormValidation/FormValidation.jsx

```diff
@@ -0,0 +1,186 @@
+import { useCallback, useState } from 'react';
+import './FormValidation.css';
+import { InputValidate, Button } from '@components';
+import { BUTTON_VARIANTS } from '@constants';
+import { Validation } from '@helpers';
+
+const FormValidation = (props) => {
+  const { handleShowForm } = props;
+  const [formData, setFormData] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+    gender: 'male',
+  });
+
+  const [errors, setErrors] = useState({
+    username: '',
+    avatar: '',
+    email: '',
+    phonenumber: '',
+    description: '',
+    address: '',
+  });
+
+  const handleChange = useCallback((e) => {
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [e.target.name]: e.target.value,
+    }));
+  }, []);
+
+  const handleBlur = useCallback((e) => {
+    const { name, value } = e.target;
+    const errorMessage = Validation(name, value);
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      [name]: errorMessage,
+    }));
+  }, []);
+
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault();
+
+      const newErrors = {};
+      for (const field in formData) {
+        const errorMessage = Validation(field, formData[field]);
+        if (errorMessage) {
+          newErrors[field] = errorMessage;
+        }
+      }
+      setErrors(newErrors);
+
+      if (Object.keys(newErrors).length === 0) {
```

Comment:
what do you mean for this code

Reply:
At first, I create a variable newErrors with empty object
It will store any error message when Validation function loop through each input field

After that, I use Object.key to retrieve an array of strings representing all the keys in an object
Check if the array's length is 0, the form can submit